src/Tasks/LegalCNF/Config.hs

```diff
@@ -97,7 +97,7 @@ checkLegalCNFConfig LegalCNFConfig{cnfConfig = cnfConf@CnfConfig {baseConf = Bas
       = reject $ do
         english "minClauseAmount is too large. The external generator cannot generate a CNF."
         german "minClauseAmount ist zu groß. Es kann keine passende Cnf geriert werden."
-    | minStringSize < max 1 minClauseAmount * ((minClauseLength - 1) * 5 + 1) = reject $ do
+    | minStringSize < max 1 (minClauseLength + 1) * (minClauseAmount - 1) = reject $ do
```

Comment:
This would now mean that a configuration with `minClauseAmount = 2`, `minClauseLength = 2` and `minStringSize = 3` is accepted. That is, a lecturer can make that choice. What I would like to tell that lecturer is that it makes more sense to increase their `minStringSize`. After all, any real string resulting from at least two clauses with at least two literals will be much longer. The shortest being something like `"(A ∨ B) ∧ (C ∨ D)"`, which has length `17`. So, while a configuration with `minClauseAmount = 2`, `minClauseLength = 2` and `minStringSize = 3` is not *wrong* per se, it is actually not very reasonable.

A lecturer who thinks that a string with length `3` might arise in that situation probably has not really understood what the parameters stand for.

In general, the checkers (throughout all task types) serve two purposes:
- Telling the lecturer when the constraints they set are impossible to satisfy.
- Telling the lecturer when the constraints they set can be sharpened without making the search space any smaller.

So, when a lecturer sets `minClauseAmount = 2`, `minClauseLength = 2` and `minStringSize = 3`, that should be rejected and they be guided towards incresaing `minStringSize`. Since `minClauseAmount = 2`, `minClauseLength = 2` and `minStringSize = 3` has exactly the same solutions as `minClauseAmount = 2`, `minClauseLength = 2` and `minStringSize = 10` and indeed `minClauseAmount = 2`, `minClauseLength = 2` and `minStringSize = 17`, but the latter have more documentation value (someone external looking at the configuration learns more about it if `minStringSize` is higher).

Hence, and with a bit more analysis:
- The value computed here should be higher.
- But apparently the expression with `5` in it is too high.
- I bet it is an artifact of when `∨` and `∧` were still written as `\/` and `/\`. :smile:
- So simply reducing `5` to `4` might do the trick.
- Maybe the bound can be further sharpened, taking into account also the brackets and operation symbols that are used in `"( ) ∧ ( )"`.
- The message returned should probably be changed, to be more in line with what the above sketched intent is of "nudging the lecturer to higher values if that doesn't change the true search space".
- The bound for `maxStringSize` below might also benefit from reconsideration (in light of both "is it too lax?" and "is it too sharp?").
- In principle, the multipliers and addition constants (like `4`, `5`, `6` appearing now in the expressions) might even be influenced by other aspects of the configuration. (If arrow symbols are allowed in the candidate formulas, then the longest operation symbol is `<=>`, of length `3`. If no arrow symbols are allowed in the candidate formulas, then the longest operation symbol is `∨`, of length just `1`. Whether or not negation is allowed -- if that happens to be a configurable feature here -- would also change the picture. Etc.)

Reply:
Yes, dealing with this further in a separate issue makes sense. Possibly in https://github.com/fmidue/logic-tasks/issues/90.